content/doc/developer/forms/jelly-form-controls.adoc

```diff
@@ -0,0 +1,115 @@
+
+Finally, the with attribute specifies the input fields sent to the server for the validation. They are matched against the field attribute or the name attribute of other input controls. The values of the nearest input fields above the `<f:validateButton/>` are sent to the server, so this means the button has to come after the input fields. Multiple fields can be specified by using ','.
+
+On the server side, this tag invokes the standard "do"-style method like this:
+
+[source, java]
+----
+public FormValidation doTestConnection(@QueryParameter("accessId") final String accessId,
+        @QueryParameter("secretKey") final String secretKey) throws IOException, ServletException {
+    try {
+        ... do some tests ...
+        return FormValidation.ok("Success");
+    } catch (EC2Exception e) {
+        return FormValidation.error("Client error : "+e.getMessage());
+    }
+}
+----
+
+The doTestConnection method contains the validation logic. In the end, this method has to call either `FormValidation.ok`, `.warning`, or .error method. Depending on which method you use, the appropriate HTML will be rendered on the client side to indicate the status to the user.
+
+== Advanced
+
+Expandable section that shows "advanced..." button by default. Upon clicking it, a section unfolds.
+
+[source, xml]
+----
+<f:section title="Advanced Project Options">
+  <f:advanced>
+    <st:include page="configure-advanced.jelly" />
+  </f:advanced>
+</f:section>
+----
+
+== OptionalBlock
+Foldable block expanded when the menu item is checked.
+
+[source, xml]
+----
+<f:block>
+  <table>
+    <f:optionalBlock name="dynamic" title="Use existing dynamic view">
+      <f:entry title="View drive">
+        <f:textbox name="drive" value="${it.drive}"/>
+      </f:entry>
+    </f:optionalBlock>
+  </table>
+</f:block>
+----
+
+Unchecked the text box will not be displayed
+Checked the text box is displayed 
+
+== Select (drop-down menu)
+Use an `<f:entry>` tag to enclose the normal select tag.
+
+[source, xml]
+----
+<f:entry name="goalType" title="Choose Goal Type" field="goalType">
+    <select name="goalType">
+        <option value="buildGoal">Build Goal</option>
+        <option value="findBugsGoal">FindBugs goal</option>
+    </select>
+</f:entry>
+----
+
+== Select (drop-down menu) with model filled values
+Basically the same as above. You need to define the following method in the descriptor of your Describable instance:
+
+[source, java]
+----
+public ListBoxModel doFillGoalTypeItems() {
+    ListBoxModel items = new ListBoxModel();
+    for (BuildGoal goal : getBuildGoals()) {
```

Comment:
Depending on what the source of the data is, needs a permission check. Perhaps a less trivial example that returns e.g. a list of job names would make sense for that?

Reply:
it's now super trivial